pkg/datapath/linux/config/config.go

```diff
@@ -449,6 +449,11 @@ func (h *HeaderfileWriter) writeTemplateConfig(fw *bufio.Writer, e datapath.Endp
 		ctmap.WriteBPFMacros(fw, nil)
 	}
 
+	// The IPv4 gateway of the endpoint.
+	// While IPV4_GATEWAY can change when cilium agent reloads, LXC_IPV4_GATEWAY
+	// is fixed with the value at the moment the endpoint was created.
+	fmt.Fprintf(fw, "#define LXC_IPV4_GATEWAY %#x\n", byteorder.HostSliceToNetwork(e.GetIPv4Gateway(), reflect.Uint32).(uint32))
```

Comment:
Can it be that `e.GetIPv4Gateway()` has not been set yet when this is executed? It would likely explain the failures in CI.

Reply:
Indeed. When restoring endpoints from a previous version, IPv4Gateway/LXC_IPV4_GATEWAY is not set. I did not execute e2e tests because I was having local issues with vagrant/virtualbox so I tried my luck :(

My initial idea to fix #10574 was not to respond for ARP requests for the endpoint IP (LXC_IPV4). And it certainly would be enough for that specific case and simpler, but still not quite correct I guess.  So the options I see is:

1) not respond for ARP requests for the endpoint IP (LXC_IPV4) and leave it at that.
2) not respond for ARP requests for the endpoint IP (LXC_IPV4) if LXC_IPV4_GATEWAY is not available, otherwise just respond to LXC_IPV4_GATEWAY.

Option 2 hopes for a future where the old behavior could be removed completely, but probably not worth it given the potential situation of two endpoints behaving differently, even if unlikely. 

Any thoughts?

Thanks for the review!

Reply:
By the way, I since tracked my vagrant/virtualbox issue to https://github.com/hashicorp/vagrant/issues/11403, so I guess it is advised to hold any vbox upgrades until a future vagrant version becomes available.

Reply:
If it helps at all, I sometimes avoid the whole vagrant/virtualbox madness by using microk8s locally for testing purposes. This case doesn't seem like it needs multi-node, so that could be an option.